src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.cs

```diff
@@ -79,6 +80,11 @@ private RetargetingTypeParameterSymbol CreateRetargetingTypeParameter(Symbol sym
             return new RetargetingTypeParameterSymbol(this, (TypeParameterSymbol)symbol);
         }
 
+        private NativeIntegerTypeSymbol CreateRetargetingNativeInteger(Symbol symbol)
+        {
+            return new NativeIntegerTypeSymbol((NamedTypeSymbol)symbol);
```

Comment:
>return new NativeIntegerTypeSymbol((NamedTypeSymbol)symbol); [](start = 12, length = 60)

This doesn't feel right. The helpers above are for declarations in source from a referenced compilation. Native int types are not.

Reply:
Also, I think this won't give us the right behavior when underlying type is missing, for example. We are not supposed to create ```NativeIntegerTypeSymbol``` instances in cases like that. Please add a test for this scenario with appropriate symbol validation.

---
In reply to: [393226039](https://github.com/dotnet/roslyn/pull/42419#discussion_r393226039) [](ancestors = 393226039)

Reply:
> Please add a test for this scenario 

To clarify, the scenario is the types are not missing in compilation #1, but are missing in compilation #2 (the core library is not referenced by compilation #2, or it doesn't have them).

---
In reply to: [393228651](https://github.com/dotnet/roslyn/pull/42419#discussion_r393228651) [](ancestors = 393228651,393226039)